pkg/reconciler/artifactbuildrequest/artifactbuildrequest.go

```diff
@@ -120,106 +94,55 @@ func (r *ReconcileArtifactBuildRequest) Reconcile(ctx context.Context, request r
 			return reconcile.Result{}, err
 		}
 		if len(list.Items) == 0 {
-			//no TR found, this is odd
-			//I guess just go back to new
-			abr.Status.State = v1alpha1.ArtifactBuildRequestStateNew
-			return reconcile.Result{Requeue: true}, r.client.Update(ctx, &abr)
-		}
-		tr := list.Items[0]
-		if tr.Status.CompletionTime != nil {
-			//make sure the tr is done
-			condition := tr.Status.GetCondition(apis.ConditionSucceeded)
-			if condition.IsTrue() {
-				abr.Status.State = v1alpha1.ArtifactBuildRequestStateBuilding
-				//now lets create the dependency build object
-				//once this object has been created it's resolver takes over
-				var scmUrl string
-				var scmTag string
-				var scmType string
-				var message string
-				var path string
+			//no existing build object found, lets create one
+			db := &v1alpha1.DependencyBuild{}
+			db.Namespace = abr.Namespace
+			db.Labels = lbls
+			//TODO: name should be based on the git repo, not the abr, but needs
+			//a sanitization algorithm
+			db.GenerateName = abr.Name + "-"
```

Comment:
switching to generate name is new, right?

not a bad thing, but I'm curious why you pivoted to that.  Can you elaborate?

Reply:
Nope, it was moved from here: https://github.com/redhat-appstudio/jvm-build-service/pull/35/files#diff-512dbf53bb69666cabbc97bb8bc3bd9caa90fbfedaac29c57c14101c513d5e68L186

generatedName is used because ATM this name is not guaranteed to be stable, it depends on which ABR generates the build. If we had a way of generating human readable stable names from SCM URI + context path that meet the kube naming rules then we could use that, but it did not seem worth spending time on at this point.